virttest/env_process.py

```diff
@@ -1073,11 +1073,11 @@ def preprocess(test, params, env):
     # change the kernel params if configured
     if kernel_extra_params_add or kernel_extra_params_remove:
         for vm in env.get_all_vms():
-            if vm:
-                if not vm.is_alive():
+            if params.get("vm_type") == "libvirt":
+                if vm and not vm.is_alive():
                     vm.start()
-                utils_test.update_boot_option(vm, args_added=kernel_extra_params_add,
-                                              args_removed=kernel_extra_params_remove)
+            utils_test.update_boot_option(vm, args_added=kernel_extra_params_add,
+                                          args_removed=kernel_extra_params_remove)
```

Comment:
I thought there doesn't need to start VM after  `process(test, params, env, preprocess_image, preprocess_vm)` that will not disturb the status of start_vm of the case level, should keep the initial status of VM.
Thanks.

Reply:
do you mean even for libvirt test, it is not necessary to start the VM as preprocess_vm takes care ?

Reply:
umr~~
yes,  from my point of view.

Reply:
Updating the boot options remains the same. the catch here is to handle managing the VM in generic way.

Reply:
One scenario that if the parameter `start_vm` is set  `start_vm = no` in the case level, then set some parameters for VM by preprocess_vm.But the VM has been started at this time, so you have to shut down it then preprocess_vm again. I think it goes against the purpose of the case.

Reply:
if `start_vm = no` is set by the test case then test should take care to update kernel param if at all it is going to use VM, so we can raise `TestSetupError()` if user sets both `start_vm = no` and `kernel_extra_params_*` from env_process. what say ?